### PR TITLE
ramda: Add R.pathSatisfies

### PR DESCRIPTION
**Pull request by afharo**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: http://ramdajs.com/docs/#pathSatisfies
- [X] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/ramda/index.d.ts [hunk 1]
```diff
@@ -1,65 +1,53 @@
...
 declare namespace R {
     type Ord = number | string | boolean;
 
-    type Path = (number | string)[];
+    type Path = Array<(number | string)>;
 
-    interface ListIterator<T, TResult> {
-        (value: T, index: number, list: T[]): TResult;
-    }
+    type ListIterator<T, TResult> = (value: T, index: number, list: T[]) => TResult;
```
Comment (minestarks) [inline types/ramda/index.d.ts]: This type isn't used anywhere.

File: types/ramda/index.d.ts [hunk 2]
```diff
@@ -1,65 +1,53 @@
...
 
     interface Functor<T> {
         map<U>(fn: (t: T) => U): Functor<U>;
     }
 
-    interface ObjectIterator<T, TResult> {
-        (element: T, key: string, obj: Dictionary<T>): Dictionary<TResult>;
-    }
+    type ObjectIterator<T, TResult> = (element: T, key: string, obj: Dictionary<T>) => Dictionary<TResult>;
```
Comment (minestarks) [inline types/ramda/index.d.ts]: Same here, it's not used anywhere

File: types/ramda/index.d.ts [hunk 3]
```diff
@@ -1,65 +1,53 @@
...
 
-    interface KeyValuePair<K, V> extends Array<K | V> { 0 : K; 1 : V; }
+    interface KeyValuePair<K, V> extends Array<K | V> { 0: K; 1: V;
+    }
```
Comment (minestarks) [inline types/ramda/index.d.ts]: Fix weird newline please

**Review by minestarks — CHANGES_REQUESTED** [review]

File: types/ramda/index.d.ts (hunk 1 above)
Comment (afharo) [inline types/ramda/index.d.ts]: You are right! Removed!

File: types/ramda/index.d.ts (hunk 2 above)
Comment (afharo) [inline types/ramda/index.d.ts]: You are right! Removed!

File: types/ramda/index.d.ts (hunk 3 above)
Comment (afharo) [inline types/ramda/index.d.ts]: Fixed

**Comment by afharo** [comment]
@minestarks Fixed your comments (thank you, btw). Can you have a second look?

**Review by minestarks — APPROVED** [review]

**Comment by afharo** [comment]
Can these changes be npm published?
It looks like there was an `npm publish` just after the merge, but if I update my package, my changes are still missing :(

**Comment by minestarks** [comment]
@afharo I see them now.

**Comment by afharo** [comment]
@minestarks yeah! now working! thank you!

**Pull request merged**